src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ModifyQueueCapacityAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotClusterSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.QueueRejectionClusterRca;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// This is a sample decider implementation to finalize decision maker interfaces.
+// TODO: 1. Read action priorities from a configurable yml
+// TODO: 2. Read current queue capacity from NodeConfigurationRca (PR #252)
+
+public class QueueHealthDecider extends Decider {
+
+  public static final String NAME = "queue_health";
+
+  private QueueRejectionClusterRca queueRejectionRca;
+  List<String> actionsByUserPriority = new ArrayList<>();
+  private int counter = 0;
+
+  public QueueHealthDecider(long evalIntervalSeconds, int decisionFrequency, QueueRejectionClusterRca queueRejectionClusterRca) {
+    // TODO: Also consume NodeConfigurationRca
+    super(evalIntervalSeconds, decisionFrequency);
+    this.queueRejectionRca = queueRejectionClusterRca;
+    configureActionPriority();
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    Decision decision = new Decision(System.currentTimeMillis(), NAME);
+    counter += 1;
+    if (counter < decisionFrequency) {
+      return decision;
+    }
+
+    counter = 0;
+    if (queueRejectionRca.getFlowUnits().isEmpty()) {
+      return decision;
+    }
+
+    HotClusterSummary clusterSummary = queueRejectionRca.getFlowUnits().get(0).getSummary();
```

Comment:
`queueRejectionRca` can be empty in some cases, which means that the summary might be `null`. We might want to make an early return in that case ?